static/gsApp/components/quotaExceededNavItem.tsx

```diff
@@ -0,0 +1,292 @@
+  const [isDismissed, setIsDismissed] = useState(false);
+  const [hasLoaded, setHasLoaded] = useState(false);
+
+  useEffect(() => {
+    if (!subscription.categories) {
+      return;
+    }
+    const result = sortCategoriesWithKeys(subscription?.categories ?? {})
+      .filter(
+        ([category]) =>
+          category !== DataCategory.SPANS_INDEXED ||
+          subscription?.hadCustomDynamicSampling
+      )
+      .reduce((acc, [category, currentHistory]) => {
+        if (currentHistory.usageExceeded) {
+          acc.push(category);
+        }
+        return acc;
+      }, [] as string[]);
+    setExceededCategories(result);
+    setPromptsToCheck(
+      result
+        .map(category => {
+          const categoryInfo = getCategoryInfoFromPlural(category as DataCategory);
+          if (categoryInfo) {
+            return `${categoryInfo.snakeCasePlural ?? category}_overage_alert`;
+          }
+          return null;
+        })
+        .filter(Boolean) as string[]
+    );
+  }, [subscription.categories, subscription.hadCustomDynamicSampling]);
+
+  const {
+    data: prompts,
+    isPending,
+    isError,
+  } = usePromptsCheck(
+    {
+      organization,
+      feature: promptsToCheck,
+    },
+    {staleTime: 0, enabled: promptsToCheck.length > 0}
+  );
+
+  const promptIsDismissedForBillingPeriod = useCallback(
+    (prompt: {dismissed_ts?: number; snoozed_ts?: number}) => {
+      const {snoozed_ts, dismissed_ts} = prompt || {};
+      const time = snoozed_ts || dismissed_ts;
+      if (!time) {
+        return false;
+      }
+      const onDemandPeriodEnd = new Date(subscription.onDemandPeriodEnd);
+      onDemandPeriodEnd.setHours(23, 59, 59);
+      return time <= onDemandPeriodEnd.getTime() / 1000;
+    },
+    [subscription.onDemandPeriodEnd]
+  );
+
+  const {
+    isOpen,
+    triggerProps: overlayTriggerProps,
+    overlayProps,
+    state: overlayState,
+  } = usePrimaryButtonOverlay({});
+  const theme = useTheme();
+  const api = useApi();
+  const prefersStackedNav = usePrefersStackedNav();
+
+  useEffect(() => {
+    if (!hasLoaded && prompts?.features) {
+      if (
+        promptsToCheck.every(prompt => {
+          const promptData = prompts.features?.[prompt];
+          return promptData && promptIsDismissedForBillingPeriod(promptData);
+        })
+      ) {
+        setIsDismissed(true);
+      }
+      setHasLoaded(true);
+    }
+  }, [hasLoaded, prompts?.features, promptsToCheck, promptIsDismissedForBillingPeriod]);
+
+  const shouldShow =
+    prefersStackedNav &&
+    exceededCategories.length > 0 &&
+    !subscription.hasOverageNotificationsDisabled;
+  if (!shouldShow || isPending || isError) {
+    return null;
+  }
+
+  const animateProps: MotionProps = {
+    animate: {
+      rotate: [0, -15, 15, -15, 15, -15, 0],
+      scale: [1, 1.25, 1.25, 1.25, 1.25, 1.25, 1],
+    },
+    transition: {
+      duration: 0.7,
+      repeat: Infinity,
```

Comment:
Do we really want this to repeat infinitely?

Reply:
😬 i will ask the team to confirm!

Reply:
In other words, do we want to ~~annoy~~ alert the customer until they pay us more money  😆😅